### ISSUE TITLE
About control signal prediction

### ISSUE THREAD

**Issue by TonyXuQAQ** [jxbbb/ADAPT#17]
Thanks for sharing the code. Recently I read your excellent work ADAPT, and I have some questions about the control prediction part.

1. I wonder whether you use past control signals for control signal prediction, i.e., use a_{t-1}, a_{t-2},....,a_{0} for the prediction of a_t. Or you directly predict control signals from input video images.

2. If you directly predict controls from input videos, since you do not have time length of each video clip as input, how could you predict speed from the input video? Or say, how could you know whether the input video is 3 seconds or 10 seconds? It seems that you do not send this information into your network.

**Comment by jxbbb** [comment]
We thank a lot for these constructive suggestions and comments.
1. This is a good suggestion. But in our implement, we use a transformer encoder to directly predict control signals from input video images​. A transformer decoder may work better.
2. This is a good question, and we find this is a bug for adapt, where we ignore the fps of the video after sampling. Thus a good solution is to add an temporal embedding that utilize fps as input. We would fix this bug soon.​

**Comment by LiuPengJX** [comment]
> We thank a lot for these constructive suggestions and comments.
> 
> 1. This is a good suggestion. But in our implement, we use a transformer encoder to directly predict control signals from input video images​. A transformer decoder may work better.
> 2. This is a good question, and we find this is a bug for adapt, where we ignore the fps of the video after sampling. Thus a good solution is to add an temporal embedding that utilize fps as input. We would fix this bug soon.​

Hello, the uniform sampling method mentioned in paper, does fps calculation use video duration divided by 2，4，8，16 or 32

**Comment by LiuPengJX** [comment]
> We thank a lot for these constructive suggestions and comments.
> 
> 1. This is a good suggestion. But in our implement, we use a transformer encoder to directly predict control signals from input video images​. A transformer decoder may work better.
> 2. This is a good question, and we find this is a bug for adapt, where we ignore the fps of the video after sampling. Thus a good solution is to add an temporal embedding that utilize fps as input. We would fix this bug soon.​

Hello, the uniform sampling method used in paper, does fps calculation use video duration divided by 2，4，8，16 or 32

**Comment by jxbbb** [comment]
Yes, this right and you can also use video duration as input. But this bdd-x is too small and it is easy to overfit. Thus I suggest doing cropping on the duration of the video to make the video more consistent.
